back-end/API/app/services/auth_service.py

```
@@ -0,0 +1,78 @@
+# TODO quand on met le mauvais mot de passe, ca dit compte desactive
+# Reset password, la route n'existe plus :(
+class AuthService:
+    def login(self, dto: UserLoginDTO):
+        try:
+            user = user_repo.find_by_email(dto.email)
+            if not user.active:
+                raise LoginException(True, f"Impossible de se connecter avec l'email: {dto.email}")
+            hasher.verify(user.password, dto.password)
+            return self._generateToken(user), user
+        except VerifyMismatchError:
+            raise LoginException(f"Login attempt failed on user: {dto.email} invalid password")
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)